batch/batch.go

```diff
@@ -112,13 +118,35 @@ func (m *batchManager) loadExistingBatches() error {
 			m.handleBatchJobsCompleted(b, map[string]bool{b.Id: true})
 		}
 	}
-
+	util.Infof("Loaded %d batches", len(m.Batches))
 	return nil
 }
 
+func (m *batchManager) lockBatchIfExists(batchId string) {
+	m.mu.Lock()
+	if _, ok := m.Batches[batchId]; !ok {
+		m.mu.Unlock()
+		return
+	}
+	lock := m.batchMu[batchId]
+	m.mu.Unlock()
+	lock.Lock()
```

Comment:
Same here, why don't you add `defer m.mu.Unlock()` below `m.mu.Lock()`. It'll be safer than explicitly calling `m.mu.Unlock()` in several places.